src/bun.js/webcore.zig

```diff
@@ -13,6 +14,279 @@ pub const Lifetime = enum {
+        };
+
+        // 5. Invoke WebDriver BiDi user prompt opened with this, "confirm", and message.
+        // *  Not relevant in a server context.
+
+        // 6. Pause until the user responds either positively or negatively.
+        var stdin = std.io.getStdIn();
+        var reader = stdin.reader();
+
+        const first_byte = reader.readByte() catch {
+            return JSC.JSValue.jsBoolean(false).asObjectRef();
+        };
+
+        // 7. Invoke WebDriver BiDi user prompt closed with this, and true if
+        //    the user responded positively or false otherwise.
+        // *  Not relevant in a server context.
+
+        switch (first_byte) {
+            '\n' => return JSC.JSValue.jsBoolean(false).asObjectRef(),
+            'y' => {
+                const next_byte = reader.readByte() catch {
+                    // They may have said yes, but the stdin is invalid.
+                    return JSC.JSValue.jsBoolean(false).asObjectRef();
+                };
+
+                if (next_byte == '\n') {
+                    // 8. If the user responded positively, return true;
+                    //    otherwise, the user responded negatively: return false.
+                    return JSC.JSValue.jsBoolean(true).asObjectRef();
+                }
+            },
+            else => {},
+        }
+
+        while (reader.readByte()) |b| {
+            if (b == '\n') break;
+        } else |_| {}
+
+        // 8. If the user responded positively, return true; otherwise, the user
+        //    responded negatively: return false.
+        return JSC.JSValue.jsBoolean(false).asObjectRef();
+    }
+};
+
+pub const Prompt = struct {
+    pub const Class = JSC.NewClass(
+        void,
+        .{ .name = "prompt" },
+        .{
+            .@"call" = .{ .rfn = call },
+        },
+        .{},
+    );
+
+    /// Adapted from `std.io.Reader.readUntilDelimiterArrayList` to only append.
+    pub fn readUntilDelimiterArrayListAppend(
+        reader: anytype,
+        array_list: *std.ArrayList(u8),
+        delimiter: u8,
+        max_size: usize,
+    ) !void {
+        while (true) {
+            if (array_list.items.len == max_size) {
+                return error.StreamTooLong;
+            }
+
+            var byte: u8 = try reader.readByte();
+
+            if (byte == delimiter) {
+                return;
+            }
+
+            try array_list.append(byte);
+        }
+    }
+
+    /// https://html.spec.whatwg.org/multipage/timers-and-user-prompts.html#dom-prompt
+    pub fn call(
+        // this
+        _: void,
+        ctx: JSC.C.JSContextRef,
+        // function
+        _: JSC.C.JSObjectRef,
+        // thisObject
+        _: JSC.C.JSObjectRef,
+        arguments: []const JSC.C.JSValueRef,
+        _: JSC.C.ExceptionRef,
+    ) JSC.C.JSValueRef {
+        var stdout = std.io.getStdOut();
+        const has_message = arguments.len != 0 and arguments[0] != null;
```

Comment:
`arguments.len` is all we need to check if a JSValue is null. Technically they can be null too but the arguments buffer wouldn't have them